### ISSUE TITLE
Ref with number in parentheses doesn't work

### ISSUE THREAD

**Issue by davystrong** [typst/typst#4721]
### Description

If I have a citation key that starts with a number (say `1something`), I can't `ref` it in parentheses. Specifically (assuming each of these keys is actually defined):

```
@1something // Works. Outputs one ref node
(@something // Works. Outputs one [(] node and one ref node
(@1something // Doesn't work. Just outputs as a single text node
```

In reality, of course, the parentheses would normally be closed after the citation, but this is the minimal case.

The problem can be solved by adding something like this to the top of the document:

```
#show regex(".*\(@[A-Za-z\d_\-:.]+.*"): node => {
  let match = node.text.match(regex("(.*\()@([A-Za-z\d_\-:.]+)(.*)"))
  [#match.captures.at(0)]
  ref(label(match.captures.at(1)))
  [#match.captures.at(2)]
}
```
I imagine this will cause other parse errors, but it works for me in the meantime. This doesn't solve syntax highlighting in VS Code.

### Operating system

macOS

### Typst version

- [X] I am using the latest version of Typst

**Comment by EpicEricEE** [comment]
Duplicate of #4536, should be fixed on main with #4553.

**Comment by davystrong** [comment]
Thanks! I searched, but couldn't find that.
